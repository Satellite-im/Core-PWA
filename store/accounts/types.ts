export enum RegistrationStatus {
  UKNOWN = 'unknown',
  IN_PROGRESS = 'in_progress',
  FUNDING_ACCOUNT = 'funding_account',
  SENDING_TRANSACTION = 'sending_transaction',
  REGISTERED = 'registered',
}

export interface AccountsState {
  loading?: boolean
  locked: boolean
  pin?: string
  pinHash: string
  active: string
  gasPrice: string
  phrase: string
  error: string
  encryptedPhrase: string
  registered: boolean
  registrationStatus: RegistrationStatus
}

export enum AccountsError {
  INVALID_PIN = 'errors.accounts.invalid_pin',
  PIN_TOO_SHORT = 'errors.accounts.pin_too_short',
  UNABLE_TO_CREATE_MNEMONIC = 'errors.accounts.unable_to_create_mnemonic',
  MNEMONIC_NOT_PRESENT = 'errors.accounts.mnemonic_not_present',
  USER_DERIVATION_FAILED = 'errors.accounts.user_derivation_failed',
  USER_NOT_REGISTERED = 'errors.accounts.user_not_registered',
  USER_ALREADY_REGISTERED = 'errors.accounts.user_already_registered',
  PAYER_NOT_PRESENT = 'errors.accounts.payer_not_present',
}

export interface UserRegistrationPayload {
  name: string
  photoHash: string
  status: string
}
