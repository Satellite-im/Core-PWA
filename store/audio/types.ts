export interface AudioState {
  previousVolume: Number
  inputVolume: Number
  muted: Boolean
  volume: Number
  deafened: Boolean
  sounds: {
    inboundMedia: Number
    outboundMedia: Number
    system: Number
  }
}
