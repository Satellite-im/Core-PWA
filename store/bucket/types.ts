export interface BucketState {
  initialized: boolean
}
