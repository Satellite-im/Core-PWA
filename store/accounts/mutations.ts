import { AccountsState, RegistrationStatus } from './types'
import { User } from '~/libraries/Iridium/users/types'

const mutations = {
  setPin(state: AccountsState, pin: string) {
    state.pin = pin
  },
  setStorePin(state: AccountsState, storePin: boolean) {
    state.storePin = storePin
  },
  setRegistry(state: AccountsState, registry: boolean) {
    state.registry = registry
  },
  setPinHash(state: AccountsState, pinHash: string) {
    state.pinHash = pinHash
  },
  unlock(state: AccountsState, pin: string) {
    state.locked = false
    state.pin = pin
  },
  lock(state: AccountsState) {
    state.locked = true
  },
  setEncryptedPhrase(state: AccountsState, encryptedPhrase: string) {
    state.encryptedPhrase = encryptedPhrase
  },
  setPhrase(state: AccountsState, decryptedPhrase: string) {
    state.phrase = decryptedPhrase
  },
  setActiveAccount(state: AccountsState, activeAccountPubkey: string) {
    state.active = activeAccountPubkey
    state.initialized = true
  },
  setProfilePicture(state: AccountsState, image: string) {
    if (state.details) {
      state.details.photoHash = image
    }
  },
  setUserDetails(state: AccountsState, details: User) {
    state.details = {
      did: details.did,
      name: details.name,
      status: details.status,
      photoHash: details.photoHash,
    }
    state.welcomePopup = true
  },
  dismissWelcome(state: AccountsState) {
    state.welcomePopup = false
  },
  setUserPeerId(state: AccountsState, peerId: string) {
    if (state.details) {
      state.details = {
        ...state.details,
        peerId,
      }
    }
  },
  setRegistrationStatus(
    state: AccountsState,
    registrationStates: RegistrationStatus,
  ) {
    state.registrationStatus = registrationStates
  },
  setLastVisited(state: AccountsState, lastVisited: string) {
    state.lastVisited = lastVisited
  },
  setAdapter(state: AccountsState, adapter: string) {
    state.adapter = adapter
  },
  setEntropy(state: AccountsState, entropyMessage: string) {
    state.entropyMessage = entropyMessage
  },
}

export default mutations
