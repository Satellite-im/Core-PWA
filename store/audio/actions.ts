import type { AudioState } from './types'
import { Sounds } from '~/libraries/SoundManager/SoundManager'
import { $WebRTC } from '~/libraries/WebRTC/WebRTC'
import { ActionsArguments } from '~/types/store/store'

export default {
  /**
   * @method toggleMute
   * @description Toggles mute for outgoing audio
   * @example @click="toggleMute"
   */
  async toggleMute({
    state,
    commit,
    dispatch,
    rootState,
  }: ActionsArguments<AudioState>) {
    const { activeCall } = rootState.webrtc
    const call = activeCall && $WebRTC.getCall(activeCall.callId)

    dispatch('sounds/playSound', state.muted ? Sounds.MUTE : Sounds.UNMUTE, {
      root: true,
    })

    if (!call) {
      return
    }

    if (!state.muted) {
      await call.mute({ kind: 'audio' })
      commit('setMute', true)
      return
    }
    await call.unmute({ kind: 'audio' })
    commit('setMute', false)
  },
  /**
   * @method toggleDeafen
   * @description Toggles deafen for incoming audio
   * @example @click="toggleDeafen"
   */
  toggleDeafen({ commit, dispatch, state }: any) {
    const deafened = state.deafened
    if (!deafened) {
      dispatch('sounds/playSound', Sounds.DEAFEN, { root: true })
      dispatch('sounds/setMuteSounds', true, { root: true })
    } else {
      dispatch('sounds/setMuteSounds', false, { root: true })
      dispatch('sounds/playSound', Sounds.UNDEAFEN, { root: true })
    }
    commit('deafen')
  },
}
