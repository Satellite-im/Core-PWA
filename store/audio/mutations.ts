import { AudioState } from './types'

const mutations = {
  mute(state: AudioState) {
    state.muted = !state.muted
  },
  toggleMute(state: AudioState) {
    state.muted = !state.muted
  },
  deafen(state: AudioState) {
    const isDeafened = state.deafened
    state.previousVolume = isDeafened ? state.previousVolume : state.volume
    state.volume = isDeafened ? state.previousVolume : 0
    state.deafened = !isDeafened
  },
  setVolume(state: AudioState, volume: Number) {
    state.previousVolume = volume
    state.volume = volume
  },
  setSystemVolume(state: AudioState, systemVolume: Number) {
    state.sounds.system = systemVolume
  },
  setInputVolume(state: AudioState, inputVolume: Number) {
    state.inputVolume = inputVolume
  },
  setSoundLevel(
    state: AudioState,
    sound: {
      inboundMedia: Number
      outboundMedia: Number
      system: Number
    },
  ) {
    state.sounds = sound
  },
}

export default mutations
