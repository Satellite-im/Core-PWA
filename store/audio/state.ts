interface AudioState {
  muted: Boolean
  deafened: Boolean
  volume: Number
  inputVolume: Number
  previousVolume: Number
  sounds: {
    inboundMedia: Number
    outboundMedia: Number
    system: Number
  }
}

const InitalAudioState = (): AudioState => ({
  muted: false,
  deafened: false,
  volume: 100,
  inputVolume: 100,
  previousVolume: 0,
  sounds: {
    inboundMedia: 100,
    outboundMedia: 100,
    system: 100,
  },
})

export default InitalAudioState
