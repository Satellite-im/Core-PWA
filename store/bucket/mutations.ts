import { BucketState } from './types'
import { FilSystem } from '~/libraries/Files/FilSystem'

const mutations = {
  bucketInitialized(state: BucketState, status: boolean) {
    state.initialized = status
  },
  updateFileSystem(state: BucketState, fs: FilSystem) {
    state.fileSystem = fs
  },
  openDirectory(state: BucketState, name: string) {
    state.fileSystem.openDirectory(name)
  },
  createDirectory(state: BucketState, name: string) {
    state.fileSystem.createDirectory(name)
  },
  goBackToDirectory(state: BucketState, name: string) {
    state.fileSystem.goBackToDirectory(name)
  },
  createFile(state: BucketState, name: string) {
    state.fileSystem.createFile(name)
  },
}

export default mutations
