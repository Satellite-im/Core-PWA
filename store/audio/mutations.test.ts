import * as mutations from '~/store/audio/mutations'
import InitialAudioState from '~/store/audio/state'
import { AudioState } from '~/store/audio/types'

let instance: any
let initialState: AudioState

describe('Test audio/mutations', () => {
  beforeEach(() => {
    instance = mutations.default
    initialState = InitialAudioState()
  })

  test('setMute to true', () => {
    const argument = true
    instance.setMute(initialState, argument)
    expect(initialState.muted).toBeTruthy()
  })

  test('setMute to false', () => {
    const argument = false
    instance.setMute(initialState, argument)
    expect(initialState.muted).toBeFalsy()
  })
})

describe('mutations.default.mute', () => {
  test('0', () => {
    const result: any = mutations.default.mute({
      previousVolume: -100,
      inputVolume: -1,
      muted: true,
      volume: 0,
      deafened: false,
      sounds: { inboundMedia: 0.0, outboundMedia: -29.45, system: 10.23 },
    })
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    const result: any = mutations.default.mute({
      previousVolume: -1,
      inputVolume: 100,
      muted: true,
      volume: 1,
      deafened: false,
      sounds: { inboundMedia: -1.0, outboundMedia: -0.5, system: 10.23 },
    })
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    const result: any = mutations.default.mute({
      previousVolume: 0,
      inputVolume: 1,
      muted: true,
      volume: -100,
      deafened: true,
      sounds: { inboundMedia: 0.0, outboundMedia: 0.0, system: 0.5 },
    })
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    const result: any = mutations.default.mute({
      previousVolume: 1,
      inputVolume: -100,
      muted: true,
      volume: -100,
      deafened: false,
      sounds: { inboundMedia: -1.0, outboundMedia: 1.0, system: 1.0 },
    })
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    const result: any = mutations.default.mute({
      previousVolume: -1,
      inputVolume: -1,
      muted: true,
      volume: 100,
      deafened: true,
      sounds: { inboundMedia: -29.45, outboundMedia: 0.0, system: 0.5 },
    })
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    const result: any = mutations.default.mute({
      previousVolume: NaN,
      inputVolume: NaN,
      muted: false,
      volume: NaN,
      deafened: false,
      sounds: { inboundMedia: NaN, outboundMedia: NaN, system: NaN },
    })
    expect(result).toMatchSnapshot()
  })
})

describe('mutations.default.toggleMute', () => {
  test('0', () => {
    const result: any = mutations.default.toggleMute({
      previousVolume: 100,
      inputVolume: 100,
      muted: false,
      volume: -100,
      deafened: false,
      sounds: { inboundMedia: 10.0, outboundMedia: 10.23, system: -1.0 },
    })
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    const result: any = mutations.default.toggleMute({
      previousVolume: 0,
      inputVolume: -100,
      muted: true,
      volume: -1,
      deafened: true,
      sounds: { inboundMedia: -0.5, outboundMedia: -29.45, system: 0.5 },
    })
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    const result: any = mutations.default.toggleMute({
      previousVolume: 0,
      inputVolume: -100,
      muted: true,
      volume: 0,
      deafened: false,
      sounds: { inboundMedia: 10.0, outboundMedia: 0.0, system: 0.0 },
    })
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    const result: any = mutations.default.toggleMute({
      previousVolume: -1,
      inputVolume: 0,
      muted: true,
      volume: 100,
      deafened: true,
      sounds: { inboundMedia: -29.45, outboundMedia: -0.5, system: 0.5 },
    })
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    const result: any = mutations.default.toggleMute({
      previousVolume: -100,
      inputVolume: -1,
      muted: false,
      volume: 1,
      deafened: true,
      sounds: { inboundMedia: 0.0, outboundMedia: 0.0, system: -1.0 },
    })
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    const result: any = mutations.default.toggleMute({
      previousVolume: -Infinity,
      inputVolume: -Infinity,
      muted: true,
      volume: -Infinity,
      deafened: false,
      sounds: {
        inboundMedia: -Infinity,
        outboundMedia: -Infinity,
        system: -Infinity,
      },
    })
    expect(result).toMatchSnapshot()
  })
})

describe('mutations.default.deafen', () => {
  test('0', () => {
    const result: any = mutations.default.deafen({
      previousVolume: 1,
      inputVolume: 100,
      muted: false,
      volume: 0,
      deafened: true,
      sounds: { inboundMedia: 10.0, outboundMedia: 1.0, system: 10.0 },
    })
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    const result: any = mutations.default.deafen({
      previousVolume: -1,
      inputVolume: 100,
      muted: false,
      volume: -1,
      deafened: false,
      sounds: { inboundMedia: 1.0, outboundMedia: 10.0, system: -1.0 },
    })
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    const result: any = mutations.default.deafen({
      previousVolume: -100,
      inputVolume: 1,
      muted: false,
      volume: 1,
      deafened: false,
      sounds: { inboundMedia: 1.0, outboundMedia: 0.0, system: -0.5 },
    })
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    const result: any = mutations.default.deafen({
      previousVolume: 100,
      inputVolume: 1,
      muted: false,
      volume: 100,
      deafened: true,
      sounds: { inboundMedia: 0.5, outboundMedia: -1.0, system: 10.0 },
    })
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    const result: any = mutations.default.deafen({
      previousVolume: 0,
      inputVolume: 100,
      muted: true,
      volume: -1,
      deafened: false,
      sounds: { inboundMedia: 10.0, outboundMedia: 0.0, system: -29.45 },
    })
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    const result: any = mutations.default.deafen({
      previousVolume: -Infinity,
      inputVolume: -Infinity,
      muted: true,
      volume: -Infinity,
      deafened: true,
      sounds: {
        inboundMedia: -Infinity,
        outboundMedia: -Infinity,
        system: -Infinity,
      },
    })
    expect(result).toMatchSnapshot()
  })
})

describe('mutations.default.setVolume', () => {
  test('0', () => {
    const result: any = mutations.default.setVolume(
      {
        previousVolume: 0,
        inputVolume: 100,
        muted: true,
        volume: 0,
        deafened: false,
        sounds: { inboundMedia: 10.23, outboundMedia: 0.0, system: 10.0 },
      },
      -1,
    )
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    const result: any = mutations.default.setVolume(
      {
        previousVolume: 0,
        inputVolume: 100,
        muted: true,
        volume: 1,
        deafened: true,
        sounds: { inboundMedia: 10.0, outboundMedia: 10.0, system: -29.45 },
      },
      0,
    )
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    const result: any = mutations.default.setVolume(
      {
        previousVolume: -100,
        inputVolume: 100,
        muted: true,
        volume: 100,
        deafened: true,
        sounds: { inboundMedia: -29.45, outboundMedia: -29.45, system: 10.0 },
      },
      1,
    )
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    const result: any = mutations.default.setVolume(
      {
        previousVolume: 100,
        inputVolume: 0,
        muted: true,
        volume: 100,
        deafened: true,
        sounds: { inboundMedia: -1.0, outboundMedia: 0.0, system: 10.23 },
      },
      1,
    )
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    const result: any = mutations.default.setVolume(
      {
        previousVolume: -100,
        inputVolume: -100,
        muted: false,
        volume: 0,
        deafened: true,
        sounds: { inboundMedia: 10.23, outboundMedia: 10.23, system: -1.0 },
      },
      0,
    )
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    const result: any = mutations.default.setVolume(
      {
        previousVolume: -Infinity,
        inputVolume: -Infinity,
        muted: true,
        volume: -Infinity,
        deafened: true,
        sounds: {
          inboundMedia: -Infinity,
          outboundMedia: -Infinity,
          system: -Infinity,
        },
      },
      -Infinity,
    )
    expect(result).toMatchSnapshot()
  })
})

describe('mutations.default.setSoundLevel', () => {
  test('0', () => {
    const result: any = mutations.default.setSoundLevel(
      {
        previousVolume: -1,
        inputVolume: -1,
        muted: true,
        volume: -100,
        deafened: true,
        sounds: { inboundMedia: 1.0, outboundMedia: 10.0, system: 1.0 },
      },
      { inboundMedia: 0.5, outboundMedia: 1.0, system: 10.0 },
    )
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    const result: any = mutations.default.setSoundLevel(
      {
        previousVolume: 1,
        inputVolume: 0,
        muted: false,
        volume: 0,
        deafened: true,
        sounds: { inboundMedia: 1.0, outboundMedia: 10.23, system: 10.0 },
      },
      { inboundMedia: 0.0, outboundMedia: 0.0, system: 1.0 },
    )
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    const result: any = mutations.default.setSoundLevel(
      {
        previousVolume: -1,
        inputVolume: -100,
        muted: false,
        volume: 0,
        deafened: true,
        sounds: { inboundMedia: -29.45, outboundMedia: 10.23, system: 1.0 },
      },
      { inboundMedia: 1.0, outboundMedia: 10.23, system: -1.0 },
    )
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    const result: any = mutations.default.setSoundLevel(
      {
        previousVolume: -100,
        inputVolume: -100,
        muted: false,
        volume: -1,
        deafened: true,
        sounds: { inboundMedia: 1.0, outboundMedia: 10.23, system: 10.23 },
      },
      { inboundMedia: 0.5, outboundMedia: 0.5, system: 0.5 },
    )
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    const result: any = mutations.default.setSoundLevel(
      {
        previousVolume: -1,
        inputVolume: 0,
        muted: false,
        volume: 0,
        deafened: false,
        sounds: { inboundMedia: 0.0, outboundMedia: 10.23, system: 0.5 },
      },
      { inboundMedia: 0.5, outboundMedia: 10.23, system: -0.5 },
    )
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    const result: any = mutations.default.setSoundLevel(
      {
        previousVolume: Infinity,
        inputVolume: Infinity,
        muted: false,
        volume: Infinity,
        deafened: false,
        sounds: {
          inboundMedia: Infinity,
          outboundMedia: Infinity,
          system: Infinity,
        },
      },
      { inboundMedia: Infinity, outboundMedia: Infinity, system: Infinity },
    )
    expect(result).toMatchSnapshot()
  })
})

describe('mutations.default.setSystemVolume', () => {
  test('0', () => {
    const result: any = mutations.default.setSystemVolume(
      {
        previousVolume: -1,
        inputVolume: -1,
        muted: false,
        volume: 100,
        deafened: true,
        sounds: { inboundMedia: -0.5, outboundMedia: 0.0, system: 0.5 },
      },
      1,
    )
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    const result: any = mutations.default.setSystemVolume(
      {
        previousVolume: -100,
        inputVolume: -100,
        muted: true,
        volume: 1,
        deafened: true,
        sounds: { inboundMedia: -29.45, outboundMedia: 0.5, system: 10.0 },
      },
      1,
    )
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    const result: any = mutations.default.setSystemVolume(
      {
        previousVolume: -1,
        inputVolume: -1,
        muted: true,
        volume: -1,
        deafened: false,
        sounds: { inboundMedia: 0.0, outboundMedia: -0.5, system: 0.5 },
      },
      1,
    )
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    const result: any = mutations.default.setSystemVolume(
      {
        previousVolume: -100,
        inputVolume: 1,
        muted: true,
        volume: -1,
        deafened: false,
        sounds: { inboundMedia: -1.0, outboundMedia: 10.23, system: 10.23 },
      },
      -100,
    )
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    const result: any = mutations.default.setSystemVolume(
      {
        previousVolume: -1,
        inputVolume: 100,
        muted: false,
        volume: 100,
        deafened: true,
        sounds: { inboundMedia: 10.0, outboundMedia: -1.0, system: 0.0 },
      },
      100,
    )
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    const result: any = mutations.default.setSystemVolume(
      {
        previousVolume: NaN,
        inputVolume: NaN,
        muted: false,
        volume: NaN,
        deafened: false,
        sounds: { inboundMedia: NaN, outboundMedia: NaN, system: NaN },
      },
      NaN,
    )
    expect(result).toMatchSnapshot()
  })
})

describe('mutations.default.setInputVolume', () => {
  test('0', () => {
    const result: any = mutations.default.setInputVolume(
      {
        previousVolume: 100,
        inputVolume: -1,
        muted: true,
        volume: -1,
        deafened: true,
        sounds: { inboundMedia: 10.23, outboundMedia: -0.5, system: -29.45 },
      },
      -1,
    )
    expect(result).toMatchSnapshot()
  })

  test('1', () => {
    const result: any = mutations.default.setInputVolume(
      {
        previousVolume: -1,
        inputVolume: -100,
        muted: true,
        volume: 0,
        deafened: false,
        sounds: { inboundMedia: 0.5, outboundMedia: 0.0, system: 0.5 },
      },
      -100,
    )
    expect(result).toMatchSnapshot()
  })

  test('2', () => {
    const result: any = mutations.default.setInputVolume(
      {
        previousVolume: -1,
        inputVolume: -1,
        muted: true,
        volume: -1,
        deafened: true,
        sounds: { inboundMedia: 10.0, outboundMedia: 1.0, system: -29.45 },
      },
      -100,
    )
    expect(result).toMatchSnapshot()
  })

  test('3', () => {
    const result: any = mutations.default.setInputVolume(
      {
        previousVolume: 0,
        inputVolume: 0,
        muted: false,
        volume: 0,
        deafened: false,
        sounds: { inboundMedia: -29.45, outboundMedia: 10.0, system: 1.0 },
      },
      -100,
    )
    expect(result).toMatchSnapshot()
  })

  test('4', () => {
    const result: any = mutations.default.setInputVolume(
      {
        previousVolume: -100,
        inputVolume: 100,
        muted: true,
        volume: 0,
        deafened: true,
        sounds: { inboundMedia: 10.0, outboundMedia: 1.0, system: -29.45 },
      },
      -100,
    )
    expect(result).toMatchSnapshot()
  })

  test('5', () => {
    const result: any = mutations.default.setInputVolume(
      {
        previousVolume: NaN,
        inputVolume: NaN,
        muted: true,
        volume: NaN,
        deafened: true,
        sounds: { inboundMedia: NaN, outboundMedia: NaN, system: NaN },
      },
      NaN,
    )
    expect(result).toMatchSnapshot()
  })
})
