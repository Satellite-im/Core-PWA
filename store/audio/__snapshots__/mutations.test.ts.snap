// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations.default.deafen 0 1`] = `undefined`;

exports[`mutations.default.deafen 1 1`] = `undefined`;

exports[`mutations.default.deafen 2 1`] = `undefined`;

exports[`mutations.default.deafen 3 1`] = `undefined`;

exports[`mutations.default.deafen 4 1`] = `undefined`;

exports[`mutations.default.deafen 5 1`] = `undefined`;

exports[`mutations.default.mute 0 1`] = `undefined`;

exports[`mutations.default.mute 1 1`] = `undefined`;

exports[`mutations.default.mute 2 1`] = `undefined`;

exports[`mutations.default.mute 3 1`] = `undefined`;

exports[`mutations.default.mute 4 1`] = `undefined`;

exports[`mutations.default.mute 5 1`] = `undefined`;

exports[`mutations.default.setInputVolume 0 1`] = `undefined`;

exports[`mutations.default.setInputVolume 1 1`] = `undefined`;

exports[`mutations.default.setInputVolume 2 1`] = `undefined`;

exports[`mutations.default.setInputVolume 3 1`] = `undefined`;

exports[`mutations.default.setInputVolume 4 1`] = `undefined`;

exports[`mutations.default.setInputVolume 5 1`] = `undefined`;

exports[`mutations.default.setSoundLevel 0 1`] = `undefined`;

exports[`mutations.default.setSoundLevel 1 1`] = `undefined`;

exports[`mutations.default.setSoundLevel 2 1`] = `undefined`;

exports[`mutations.default.setSoundLevel 3 1`] = `undefined`;

exports[`mutations.default.setSoundLevel 4 1`] = `undefined`;

exports[`mutations.default.setSoundLevel 5 1`] = `undefined`;

exports[`mutations.default.setSystemVolume 0 1`] = `undefined`;

exports[`mutations.default.setSystemVolume 1 1`] = `undefined`;

exports[`mutations.default.setSystemVolume 2 1`] = `undefined`;

exports[`mutations.default.setSystemVolume 3 1`] = `undefined`;

exports[`mutations.default.setSystemVolume 4 1`] = `undefined`;

exports[`mutations.default.setSystemVolume 5 1`] = `undefined`;

exports[`mutations.default.setVolume 0 1`] = `undefined`;

exports[`mutations.default.setVolume 1 1`] = `undefined`;

exports[`mutations.default.setVolume 2 1`] = `undefined`;

exports[`mutations.default.setVolume 3 1`] = `undefined`;

exports[`mutations.default.setVolume 4 1`] = `undefined`;

exports[`mutations.default.setVolume 5 1`] = `undefined`;

exports[`mutations.default.toggleMute 0 1`] = `undefined`;

exports[`mutations.default.toggleMute 1 1`] = `undefined`;

exports[`mutations.default.toggleMute 2 1`] = `undefined`;

exports[`mutations.default.toggleMute 3 1`] = `undefined`;

exports[`mutations.default.toggleMute 4 1`] = `undefined`;

exports[`mutations.default.toggleMute 5 1`] = `undefined`;
